src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpSystemProxy.cs

```diff
@@ -62,24 +65,46 @@ private HttpSystemProxy(WinInetProxyHelper proxyHelper, SafeWinHttpHandle sessio
                 {
                     // Process bypass list for manual setting.
                     string[] list = proxyHelper.ProxyBypass.Split(';');
-                    List<string> tmpList = new List<string>();
+                    _bypass = new List<Regex>(list.Length);
 
                     foreach (string value in list)
                     {
                         string tmp = value.Trim();
+                        if (tmp.Length == 0)
+                        {
+                            continue;
+                        }
                         if (tmp == "<local>")
                         {
                             _bypassLocal = true;
                             continue;
                         }
-                        if (tmp.Length > 0)
+                        try
+                        {
+                            // Escape any special characters and unescape * to get wildcard pattern match.
+                            Regex re = new Regex(Regex.Escape(tmp).Replace("\\*", ".*?") + "$",
+                                            RegexOptions.Compiled | RegexOptions.IgnoreCase | RegexOptions.CultureInvariant);
```

Comment:
What kind of strings are we getting here that we're matching against?  This might be convenient, but I'm not sure it's going to perform particularly well.

Reply:
Those are generally host names and domain name fragments. They may or may not contain wildcards. '*' is only one supported.